### DISCUSSION TITLE
Ideas about what frontend libraries/patterns we should use?

### DISCUSSION THREAD

**Discussion by grgp** [mgilangjanuar/teledrive#3] [Ideas]
Hey everyone 👋

As the title says, I'm wondering how should we build the frontend for this project. Whether it's specific libraries, or just the general pattern on how should we structure our code. And since this is a React project, how to pass data between components.

I'll totally accept anything, but just for reference my usual go-tos are:
- [Chakra UI](https://chakra-ui.com) for the component library.
- Use regular React Context and useContext to pass data around.
- Rely on [colocation](https://kentcdodds.com/blog/colocation) for [most](https://kentcdodds.com/blog/state-colocation-will-make-your-react-app-faster) things to decide where things should be placed.
- Idk that's it really. I've heard https://react-hook-form.com/ a lot but I haven't used it.

Thoughts?

**Comment by mgilangjanuar** [comment]
Cool stuff, George!

Personally, I'm open to all possibilities. Usually, I use [ant design](https://ant.design/) they also have useForm hooks 😂. This is one of my projects that uses it: [mgilangjanuar/tamagotreal](https://github.com/mgilangjanuar/tamagotreal). But, yes that library is too light and some components are complicated to customize 🤔

**Comment by Aqshola** [comment]
I think using react plus chakra ui is good, but kinda confuse choosing react hooks / redux

**Comment by mgilangjanuar** [comment]
Whoa so many front-end enthusiasts here :')

Good, we'll use Chakra as the component library 🎊

**Comment by mgilangjanuar** [comment]
Anw, for the data fetching, I prefer to use [swr](https://swr.vercel.app/). And  using [axios](https://www.npmjs.com/package/axios) for the fetcher.

**Comment by grgp** [comment]
Hey also, what should we use for routing? Just regular react-router-dom?

Edit: swr looks cool let's use that, I've usually used react-query but from what I can see they're basically the same haha.

**Reply by mgilangjanuar** [reply]
Yup, I agree with using react-router-dom.

They just release [v1.0](https://twitter.com/vercel/status/1431291914171518977) 🤣

**Reply by grgp** [reply]
Nice, I'll make a PR for installing them then.
